libs/events/events.go

```diff
@@ -168,10 +172,15 @@ func (cell *eventCell) RemoveListener(listenerID string) int {
 
 func (cell *eventCell) FireEvent(data EventData) {
 	cell.mtx.RLock()
+	var listenerCopy []EventCallback
```

Comment:
why copy?

Reply:
although, copy should be fine too

Reply:
yes， this previously can exist a deadlock， when we call add/remove event in callback. but it can work under carefully control，copy is a way to discard the burden.

I prefer the copy, it free the event user.

Reply:
I'll create a new test to show the deadlock.